### PR TITLE
Update CRA and Vue samples

### PR DESCRIPTION
**Pull request by andygup**
Update CRA to v5
Update Vue to use Vite.

### CODE REVIEW

**Review by odoe — COMMENTED** [review]
Since moving vue to vite, we should change the name of the folder and package.json to just vue or vue-sample to avoid any confusion

File: esm-samples/jsapi-create-react-app/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "@testing-library/user-event": "^13.5.0",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
-    "react-scripts": "^4.0.3",
+    "react-scripts": "5.0.0",
```
Comment (dasa) [inline esm-samples/jsapi-create-react-app/package.json:12]: Should this be `^5.0.0` for consistency?

File: esm-samples/jsapi-create-react-app/package.json [hunk 2]
```diff
@@ -38,5 +38,5 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
-  }
+  }
```
Comment (dasa) [inline esm-samples/jsapi-create-react-app/package.json:41]: Undo change

File: esm-samples/jsapi-vue-cli/package.json [hunk 3]
```diff
@@ -1,36 +1,17 @@
...
-    ],
-    "parserOptions": {
-      "parser": "babel-eslint"
-    },
-    "rules": {}
+    "@vitejs/plugin-vue": "^2.0.0",
+    "vite": "^2.7.2"
   }
-}
+}
```
Comment (dasa) [inline esm-samples/jsapi-vue-cli/package.json:17]: Set editor to add new line to end of files if possible.

**Review by dasa — CHANGES_REQUESTED** [review]

File: esm-samples/jsapi-create-react-app/package.json (hunk 1 above)
Comment (andygup) [reply esm-samples/jsapi-create-react-app/package.json:12]: 👍

File: esm-samples/jsapi-create-react-app/package.json (hunk 2 above)
Comment (andygup) [reply esm-samples/jsapi-create-react-app/package.json:41]: added a new line to the end.

File: esm-samples/jsapi-vue-cli/package.json (hunk 3 above)
Comment (andygup) [reply esm-samples/jsapi-vue-cli/package.json:17]: Done.

File: esm-samples/jsapi-create-react-app/package.json (hunk 2 above)
Comment (dasa) [reply esm-samples/jsapi-create-react-app/package.json:41]: Please revert the changes to this line

**Review by odoe — APPROVED** [review]

File: esm-samples/.metrics/4.22.2.csv [hunk 4]
```diff
@@ -1,6 +1,6 @@
 Sample,Main bundle size (MB),On-disk size (MB), On-disk files
 Angular 13.2.1,1.71,8.05,207
-CRA 4.0.3,1.74,40.86,394
-Vue 3.2.23,1.72,37.39,363
+CRA 5.0.0,1.70,27.78,419
+Vue 3.2.25,1.49,7.32,268
```
Comment (dasa) [inline esm-samples/.metrics/4.22.2.csv:4]: Some nice savings 👍🏻

**Review by dasa — APPROVED** [review]

**Pull request merged**